RELEASE-NOTES.md

```diff
@@ -1,5 +1,20 @@
 # Prefect Release Notes
 
+## Release 2.14.18
+
+### Fixes
+- Fix environment variable not accepting lists — https://github.com/PrefectHQ/prefect/pull/11722
```

Comment:
This should maybe just be "allow prefect settings to accept lists?" i think?

Reply:
update in [e5fdd2b](https://github.com/PrefectHQ/prefect/pull/11769/commits/e5fdd2bc9c96062aadea25e8c81df4948227afaa)